src/stores/advertises.js

```diff
@@ -78,6 +95,38 @@ export const useAdvertiseStore = defineStore('advertises', {
         }
       }
     },
+    async fetchAdvertiseById(campaignId) {
+      let advertise
+      if (this.getALlActiveAdvertises.length || this.getAdvertises.length) {
+        advertise =
+          this.getALlActiveAdvertises?.find((advertise) => advertise.id === campaignId) ||
+          this.getAdvertises?.find((advertise) => advertise.id === campaignId)
+      }
+      if (advertise) {
+        return advertise
+      }
```

Comment:
You can simplify this by using early returns to avoid unnecessary nesting
 
 // Attempt to find the advertise in local stores first
    const advertise = this.getALlActiveAdvertises?.find(ad => ad.id === campaignId) || this.getAdvertises?.find(ad => ad.id === campaignId);

 // And return the found advertise if it exists
    if (advertise) return advertise;